src/domain/program.ts

```diff
@@ -322,7 +322,10 @@ export class ProgramManager
 
     const { isPersistent, specs } = newProgram
 
-    const { memory, vcpus } = this.parseSpecs(specs)
+    const parsedSpecs = this.parseSpecs(specs)
+    const memory = parsedSpecs?.memory
```

Comment:
What is the point of returning undefined in `this.parseSpecs` method and here defaulting to undefined as well using the optional chaining operator?

Reply:
Also this is basically the same than doing this:
`const { memory, vcpus } = this.parseSpecs(specs) || {}`